wremnants/datasets/datagroups.py

```diff
@@ -40,12 +39,14 @@ def __init__(self, infile, combine=False):
             self.wlike = os.path.basename(self.getMetaInfo()["command"].split()[0]).startswith("mz_wlike")
 
         self.lumi = None
+        # FIXME: self.datasets is currently a data member of the inherited class, we should define it here as well
         if self.datasets and self.results:
             self.data = [x for x in self.datasets.values() if x.is_data]
             if self.data:
                 self.lumi = sum([self.results[x.name]["lumi"] for x in self.data if x.name in self.results])
         self.groups = {}
-
+        self.groupNamesPostFilter = [] # keep track of group names after any filter defined by the user
```

Comment:
I'm not a big fan of having to keep this around and update it manually

Reply:
it is managed automatically by methods that modify the list of groups, so in principle the user should never directly touch it, if that's what you meant

Reply:
Ok, I just saw a call to it in the lowpu which seemed a little scary

Reply:
Yeah, this was due to calling self.updateGroupNamesPostFilter() in datagroupsLowPU.py (the function is defined in the base datagroups class), which updates it right after the low PU groups are formed (to be more precise it was actually just initialising that list). There might be some logger.debug messages that could be dropped now that it has been tested, there are probably too many at the moment even for a debug mode.